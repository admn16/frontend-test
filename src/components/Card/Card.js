import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledCard = styled.div`
  background: #fff;
  border-width: 1px 1px 1px 1px;
  border-color: rgba(90, 122, 190, 0.14);
  border-radius: 8px;
  border-style: solid;
  box-shadow: 0px 3px 0px 0px rgba(90, 122, 190, 0.2);
  padding: 10px;

  &:hover {
    border-style: solid;
    box-shadow: 0px 10px 20px 0px rgba(90, 122, 190, 0.15);
  }
`;

const Card = ({ children }) => (
  <StyledCard>{ children }</StyledCard>
);

Card.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Card;
