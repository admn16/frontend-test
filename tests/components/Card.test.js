import React from 'react';
import renderer from 'react-test-renderer';
import Card from 'components/Card/Card';

describe('<Card/>', () => {
  const props = {
    children: (
      <p>test</p>
    ),
  };

  test('renders correctly', () => {
    const tree = renderer
      .create(<Card {...props} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
