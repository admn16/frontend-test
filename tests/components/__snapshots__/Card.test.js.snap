// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> renders correctly 1`] = `
<div
  className="Card__StyledCard-hu4gdg-0 hNyZiT"
>
  <p>
    test
  </p>
</div>
`;
